Trivago/Jade/Application/Framework/JadeBundle/Controller/JsonApiController.php

```diff
@@ -265,12 +272,13 @@ protected function getCollection($resourceName, Request $httpRequest)
      */
     protected function getEntity($resourceName, $id, Request $httpRequest)
     {
-        $this->checkAccess($resourceName, ResourceConfig::ACTION_READ);
         $request = $this->requestFactory->createEntityRequest($httpRequest->query, $resourceName, $id);
         $this->listenerManager->onGetEntityRequest($request, $resourceName);
         $repository = $this->getRepository($resourceName);
         $entity = $repository->fetchOneResource($id, $request->getRelationships());
 
+        $this->checkAccess($resourceName, ResourceConfig::ACTION_READ, $entity);
```

Comment:
I think the same applies to update entity right?

Reply:
Yes. It is for read, update and delete to check if the entity is owned by the user given in the Token